crates/topos-tce-storage/src/fullnode/mod.rs

```diff
@@ -211,6 +226,18 @@ impl WriteStore for FullNodeStore {
 }
 
 impl ReadStore for FullNodeStore {
+    fn get_source_head(&self, subnet_id: &SubnetId) -> Result<Option<SourceHead>, StorageError> {
+        Ok(self
```

Comment:
Does `subnet_id` need to be a reference? Seems like we're moving out of it anyway so perhaps it can be just `subnet_id: SubnetId` in the trait?

Reply:
We do not need ownership on `subnet_id` as `SubnetId` implement `Copy`